### PR TITLE
add TaskScheduler,TaskTuner for auto-schedule

### PR DESCRIPTION
**Pull request by CtfGo**
Add several basic structure definitions of performing auto-tune:

`TaskScheduler`： Class for scheduling tasks, it selects a task to tune once a time.
`TaskTuner`：This class is responsible for tuning a specific task, it will integrate necessary components, such as to search the optimal schedule for the task.

### CODE REVIEW

File: cinn/auto_schedule/task/task_tuner.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#include "cinn/auto_schedule/task/tune_task.h"
+#include "cinn/auto_schedule/task/tuning_options.h"
+
+namespace cinn {
+namespace auto_schedule {
+
+// This class is responsible for tuning a specific task,
+// it will integrate necessary components to search the
+// optimal schedule for the task.
+class TaskTuner {
```
Comment (zhhsplendid) [inline cinn/auto_schedule/task/task_tuner.h]: Can we change the name to TaskOptimizer or something else? `TaskTuner` is very similar in spell to `TuneTask`. I think a different name can help readers.

File: cinn/auto_schedule/task_scheduler/task_scheduler.h [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+ protected:
+  // A taskScheduler object should be created with the static function Make
+  TaskScheduler(const std::vector<TuneTask>& tasks, const Config& config);
+
+  // Select a task to tune
+  virtual int NextTaskId() = 0;
+
+  // The pointer refers to all tasks
+  const std::vector<TuneTask>* tasks_;
```
Comment (zhhsplendid) [inline cinn/auto_schedule/task_scheduler/task_scheduler.h:62]: Discussed offline, we will have a higher level abstraction to be the owner of tasks, please remember that, nothing to change now :-)

File: cinn/auto_schedule/task/task_tuner.h (hunk 1 above)
Comment (CtfGo) [reply cinn/auto_schedule/task/task_tuner.h]: That's great, i have updated the name

**Review by zhhsplendid — APPROVED** [review]
LGTM

**Pull request merged**